### PR TITLE
Comments list component added to the BookDetails screen

### PR DESCRIPTION
**Pull request by EmmanuelWLX**
## Summary
- Added CommentsList component
- Changed the BookDetails screen to show details and the comments
- Some refactoring in BookDetails children components

### Android
- Nexus 6p
commentsList-nexus6p
commentsList-nexus6p

### iOS
- iPhone 6
commentsList-iPhone6
commentsList-iPhone6

## Trello card
https://trello.com/c/lP2cpdaW/16-n-72-tabla-de-comentarios

### CODE REVIEW

File: LibraryApp/src/app/screens/BookDetails/components/CommentsList/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import { View, FlatList } from 'react-native';
+import PropTypes from 'prop-types';
+
+import styles from './styles';
+import Comment from './components/Comment';
+
+class CommentsList extends Component {
+  renderItem = ({ item }) => {
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/BookDetails/components/CommentsList/index.js]: you dont need braces here `{`

File: LibraryApp/src/app/screens/BookDetails/components/CommentsList/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    return <Comment item={item} />;
+  };
+
+  keyExtractor = ({ id }) => id;
+
+  render() {
+    const { comments } = this.props;
+    return (
+      <View style={styles.container}>
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/BookDetails/components/CommentsList/index.js]: I think you could add this style inside `FlatList`

File: LibraryApp/src/app/screens/BookDetails/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import styles from './styles';
+import CommentsList from './components/CommentsList';
+import { comments } from './comments';
+import BookDetail from './components/BookDetail';
+
+class BookDetails extends Component {
+  static navigationOptions = ({ navigation }) => {
+    const { params } = navigation.state;
+    return {
+      title: params ? params.item.title : 'Unknown title for the selected book'
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/BookDetails/index.js]: Constants for texts! In a `constants` or `strings` file if there are more than just one or two

File: LibraryApp/src/app/screens/BookDetails/index.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    };
+  };
+
+  render() {
+    const { item } = this.props.navigation.state.params;
+    const shuffledComments = shuffleArray(comments);
+    return (
+      <View style={styles.container}>
+        <View>
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/BookDetails/index.js]: Do you need this?

File: LibraryApp/src/app/screens/BookDetails/index.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+          <BookDetail item={item} />
+        </View>
+        <View style={styles.commentsContainer}>
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/BookDetails/index.js]: Cant you just use the styles in CommentList?

File: LibraryApp/src/app/screens/BookDetails/components/CommentsList/index.js (hunk 1 above)
Comment (EmmanuelWLX) [reply LibraryApp/src/app/screens/BookDetails/components/CommentsList/index.js]: i really need it :( else i need to pass item.item to Comment component

File: LibraryApp/src/app/screens/BookDetails/components/CommentsList/index.js (hunk 2 above)
Comment (EmmanuelWLX) [reply LibraryApp/src/app/screens/BookDetails/components/CommentsList/index.js]: yes i can! :D

**Pull request merged**